app/controllers/concerns/hyrax/works_controller_behavior.rb

```diff
@@ -203,7 +203,7 @@ def create_work
         @curation_concern =
           form.validate(params[hash_key_for_curation_concern]) &&
           transactions['change_set.create_work']
-          .with_step_args('work_resource.add_file_sets' => { uploaded_files: uploaded_files },
+          .with_step_args('work_resource.add_file_sets' => { uploaded_files: uploaded_files, file_set_params: params[hash_key_for_curation_concern][:file_set] },
```

Comment:
i'm slightly put off by the way `file_set_params` has to be passed through multiple layers, and then reintegrated with `uploaded_files`. do you have any thoughts about this? it feels very britle, off-hand.

i'm struggling a bit to think about better patterns. the best i've got at the moment is that maybe `desired_visibility` (or `input_parameters`/`user_parameters`?) should be added as a field on the uploaded file objects, so this data can be handled down the stack without the need to pass it around as an argument?

Reply:
That's an interesting idea.  I think that it would make sense to keep those parameters in the DB so it could be passed around outside of just request/job parameters which are more ephemeral.

Reply:
i'm going to approve and merge this as is, since all the added arguments are optional.

i still think a big refactor here is desirable, but i won't block the feature on it.